gateway/api/views/files.py

```diff
@@ -38,19 +39,132 @@
         insecure=bool(int(os.environ.get("OTEL_EXPORTER_OTLP_TRACES_INSECURE", "0"))),
     )
 )
-provider.add_span_processor(otel_exporter)
+tracer_provider.add_span_processor(otel_exporter)
 if bool(int(os.environ.get("OTEL_ENABLED", "0"))):
-    trace._set_tracer_provider(provider, log=False)  # pylint: disable=protected-access
+    trace._set_tracer_provider(  # pylint: disable=protected-access
+        tracer_provider, log=False
+    )
 
 
 class FilesViewSet(viewsets.ViewSet):
-    """ViewSet for file operations handling.
-
-    Note: only tar files are available for list and download
+    """
+    ViewSet for file operations handling.
     """
 
     BASE_NAME = "files"
 
+    # Functions methods, these will be migrated to a Repository class
+    def get_function(
+        self, user, function_title: str, provider_name: str | None
+    ) -> None:
+        """
+        This method returns the specified function.
+
+        Args:
+            user: Django user of the function that wants to get it
+            function_title (str): title of the function
+            provider_name (str | None): name of the provider owner of the function
+
+        Returns:
+            Program | None: returns the function if it exists
+        """
+
+        function = None
+        if provider_name:
+            function = self.get_provider_function(
+                provider_name=provider_name, function_title=function_title
+            )
+
+            access = False
+            if function:
+                access = self.user_has_access_to_provider_function(
+                    user=user, function=function
+                )
+            if not access:
+                function = None  # We restart the function because user has no access
+
+            return function
+
+        function = self.get_user_function(user=user, function_title=function_title)
+
+        return function
```

Comment:
```suggestion
        if not provider_name:
            return self.get_user_function(user=user, function_title=function_title)
        
        function = self.get_provider_function(
            provider_name=provider_name, function_title=function_title
        )

        if function and self.user_has_access_to_provider_function(user=user, function=function):
            return function
                
        return None
```

I see it clearer this way.

Reply:
Done in [5116a1f](https://github.com/Qiskit/qiskit-serverless/pull/1529/commits/5116a1f58d08e5688df413f5117517c25ac659e6)